sql/src/main/java/io/crate/metadata/settings/CrateSettings.java

```diff
@@ -284,9 +284,39 @@ Settings settings() {
         return referenceImplementationTree;
     }
 
+    private void buildGroupSettingReferenceTree(String prefix, Map.Entry<String, Settings> SettingEntry, Map<String, ReferenceImplementation> referenceMap) {
+        Map<String, String> settingsMap = SettingEntry.getValue().getAsMap();
+
+        if (!settingsMap.isEmpty()) {
```

Comment:
I think the different cases would require some comments, I don't completely understand them ;)